src/app/redux/priceChartSlice.ts

```diff
@@ -54,6 +103,28 @@ function convertAlphaDEXData(data: adex.Candle[]): OHLCVData[] {
   return tradingViewData;
 }
 
+//Aysnc Thunk for the update the legend using current candle when site is first loaded
+export const fetchCandlesForInitialPeriod = createAsyncThunk(
+  "priceChart/fetchCandlesForInitialPeriod",
+  async (_, { dispatch }) => {
+    const candlesMap = adex.clientState.currentPairCandlesList;
+    const ohlcvData = convertAlphaDEXData(candlesMap);
+    if (ohlcvData && ohlcvData.length > 0) {
+      const latestOHLCVData = ohlcvData[ohlcvData.length - 1];
+
+      dispatch(setLegendCandlePrice(latestOHLCVData));
+      dispatch(setLegendChange(latestOHLCVData));
+      dispatch(
+        setLegendPercChange({
+          currentOpen: latestOHLCVData.open,
+          currentClose: latestOHLCVData.close,
+        })
+      );
+      dispatch(setLegendCurrentVolume(latestOHLCVData.value));
+    }
+  }
+);
+
```

Comment:
I don't think we need to create a separate thunk for that, we are not doing any network requests here or something like this, all the data comes from adex - we need to update every time adex updates (and this is just not the same as triggering it in `useEffect` (it will most likely work because `data` variable will be updated more often than anything else, but still it's not the most straightforward way).

I suggest to create `updateAdex` in priceChartSlice reducers, accept `action: PayloadAction<adex.StaticState>` as the second param there, and update all the necessary state changes from it, this way we won't miss the updates and we won't need to call `fetch...` in the component (we already subscribe to the changes of adex state).

Reply:
i refactored the code not to use a thunk.  Important to note this is only meant to initialize the legend only when the page is first loaded or move to a different time selector.
Since the crosshair is functional and updates the legend to the candle pointed, I didn't want the legend to update to the current candle data when a new candle is printed.  It might be disruptive if a user is on a 5m timeframe and points to a certain candle.

Reply:
I don't really get the reason behind `initializeLegend`, why can't we always show what adex provides? Like you did in priceInfoSlice.

Reply:
@EvgeniiaVak I have tried the approach initialize the data with adex.clientState.currentPairCandlesList (similar to what you mentioned priceInfoSlice). The issue might be with initialization timing, the state is being initialized before the data in adex.clientState.currentPairCandlesList is populated thus making the legends all initially empty when i first load the page.

I purposely wanted to prevent updating the legend when a new candle appears that why i did not go with the approach to update the legend on state change.